src/pybind/mgr/object_format.py

```diff
@@ -142,6 +143,41 @@ def format_response(self) -> Tuple[int, str, str]:
         return -errno.EINVAL, "", f"Unsupported format: {self.format_name}"
 
 
+class ErrorResponse(ErrorResponseBase):
+    """General exception convertable to a mgr response."""
+
+    E = TypeVar("E", bound="ErrorResponse")
+
+    def __init__(self, status: str, return_value: Optional[int] = None) -> None:
+        self.return_value: int = (
+            return_value if return_value is not None else -errno.EINVAL
+        )
+        self.status: str = status
+        self.body: str = ""
+
+    def format_response(self) -> Tuple[int, str, str]:
+        return self.return_value, self.body, self.status
+
+    @property
+    def errno(self) -> int:
+        return self.return_value
+
+    def __repr__(self) -> str:
+        return f"ErrorResponse({self.status!r}, {self.return_value!r})"
+
+    @classmethod
+    def wrap(
+        cls: Type[E], exc: Exception, return_value: Optional[int] = None
+    ) -> E:
+        if return_value is None:
+            errval = getattr(exc, "errno", None)
+            if errval is not None:
+                return_value = -errval
+        err = cls(str(exc), return_value=return_value)
```

Comment:
This is a common idiom for classmethods. The classmethod supports inheritance so using `cls` rather than a specific type allows the use of the `wrap` method on subclasses of ErrorResponse.

https://docs.python.org/3/library/functions.html#classmethod